lib/core/Core.ts

```diff
@@ -109,18 +112,32 @@ export default class Core {
       Logger.warn(LogColor.yellow(`Batch writing is disabled.`));
     }
 
-    this.blockchain.startPeriodicCachedBlockchainTimeRefresh();
+    this.startPeriodicTimePull();
     this.downloadManager.start();
 
     await this.monitor.initialize(this.config, this.versionManager, this.blockchain);
   }
 
+  /**
+   * Periodically pulls time into memory to prevent db call on every request
+   */
+  private async startPeriodicTimePull () {
+    const newApproximateTime = (await this.serviceStateStore.get()).approximateTime;
+    Logger.info(`Core approximateTime updated to: ${newApproximateTime}`);
+    this.approximateTime = newApproximateTime;
+
+    // shouldContinueTimePull is only used in tests to stop the pulling
+    if (this.shouldContinueTimePull) {
+      setTimeout(async () => this.startPeriodicTimePull(), this.approximateTimeUpdateIntervalInSeconds * 1000);
+    }
+  }
+
```

Comment:
Design: can we move this logic into a separate class perhaps called `BlockchainClock`? `Core` is intentionally kept minimal to avoid becoming like another `BitcoinProcessor`.